compiler/rustc_typeck/src/check/check.rs

```diff
@@ -1098,6 +1098,28 @@ pub fn check_simd(tcx: TyCtxt<'_>, sp: Span, def_id: LocalDefId) {
                     .emit();
                 return;
             }
+
+            let len = if let ty::Array(_ty, c) = e.kind() {
+                c.try_eval_usize(tcx, tcx.param_env(def.did))
+            } else {
+                Some(fields.len() as u64)
+            };
+            if let Some(len) = len {
+                if len == 0 {
+                    struct_span_err!(tcx.sess, sp, E0075, "SIMD vector cannot be empty").emit();
+                    return;
+                } else if len > 65536 {
```

Comment:
This iimit seems fairly arbitrary looking at the code (and uses a magic number). Consider adding a comment here similar to what you have in your PR description and making this less magic, by e.g. using `u16::max_value().into()`.

Reply:
I think this also means we need to run crater? Seems like oversized SIMD arrays are fairly unlikely thing to do, but there's still a potential for a regression AFAICT.

Reply:
Good idea, made a constant in the rustc_middle that has some description attached to it.